runtime/signals.c

```diff
@@ -399,13 +399,17 @@ int caml_init_signal_stack(void)
 void caml_free_signal_stack(void)
 {
 #ifdef POSIX_SIGNALS
+  char err[1024];
   stack_t stk, disable = {0};
   disable.ss_flags = SS_DISABLE;
   /* POSIX says ss_size is ignored when SS_DISABLE is set,
      but OSX/Darwin fails if the size isn't set. */
   disable.ss_size = SIGSTKSZ;
   if (sigaltstack(&disable, &stk) < 0) {
-    caml_fatal_error_arg("Failed to reset signal stack: %s", strerror(errno));
+    if (strerror_r(errno, err, sizeof(err) != 0)) {
+      err[0] = '\0';
```

Comment:
As before, possibly handling `ERANGE`?

Reply:
Trying not to offend the original author, but: this is an unlikely failure, and the runtime system rarely bothers to include the full `strerror` explanation in its fatal errors, so I'd suggest 
```
  caml_fatal_error("Failed to reset signal stack (errno %d)", errno);
```

Reply:
Done in the latest diff to just a decimal representation.